primo/opt_model/opt_model.py

```diff
@@ -419,6 +427,38 @@ def project_max_spend_rule(model, campaign_id):
                 doc="Define limit on max project spend",
             )
             LOGGER.info("Added max project spend constraint")
+
+        def budget_constraint_slack(model):
```

Comment:
On the flexibility note, maybe include an if statement here, with an optional parameter in the "build_model" function that defaults to None, similar to the dac_budget_fraction and project_max_spend options.